pkg/virt-handler/vm.go

```diff
@@ -419,6 +418,16 @@ func (d *VirtualMachineController) updateVMIStatus(vmi *v1.VirtualMachineInstanc
 			vmi.Status.NodeName = migrationHost
 			vmi.Status.MigrationState.Completed = true
 			d.recorder.Event(vmi, k8sv1.EventTypeNormal, v1.Migrated.String(), fmt.Sprintf("The VirtualMachineInstance migrated to node %s.", migrationHost))
+			// add a condition to require a VMI time sync on the destination
+			if !condManager.HasCondition(vmi, v1.VirtualMachineInstanceRequireTimeSync) {
```

Comment:
why do we need this condition? Can't we just always try to sync the time after migration?

Reply:
Well, the thing is the destination virt handler doesn't know when the migration has ended. At least I don't see how can we know. On the other hand, the source virt-handler knows when the migration is completed. My thinking here was that we will add a condition here so the destination virt-handler will know that time should be synced when it gets the vmi.
Does it make sense?

Reply:
@vladikr as far as I remember we already have a hand-over between source- and target- virt-handler. Is that not the case? I guess as soon as the source-hander hands over the control, the target-handler can - in parallel to the normal transition tasks - set the clock. Would that work?

Reply:
@rmohr At the time that we do the hand-over, this libvirt domain is no longer "running" on the source virt-handler - the object is there but it's not active as it was already migrated to the destination. Therefore, we need the destination virt-handler to do the sync and not the source.
Does it make sense? Or are you talking about a different handover?

Reply:
Isn't that now contradicting your previous comment where you meant: 

> the thing is the destination virt handler doesn't know when the migration has ended. 

I probably mix something here ...

Reply:
Well, yes. That the reason why I added the condition during the handover on the source virt-handler. So after the hand-over when the destination virt-handler picks up the VMI it will know that the migration is over and it needs to do time sync.

Reply:
/hold

Even if we can't right now properly detect on the source that a migration to the the target node just succeeded and we need to do post-migration step (we have a pre-migration step by the way), instrumenting conditions to directly transfer commands from the source to the target is definitely not the right approach.

We should instead for instance add an additional acknowledge field to e.g. `vm.Status.MigrationState` so that the target node can distinguish between a transferred VM and not yet post-synced VM where we e.g. set the time.

Just think about other thins which we would want to do after the migration. We would need additional conditions for all of these.

@vladikr what do you think?

Reply:
Sorry @rmohr , somehow I've missed your comment... :/

I don't mind adding another field to the status. I think this is how I started this PR but then thought that a condition would be more obvious.

Thanks. I'll post an update.

Reply:
Hm, I am still not sure if that should be at all in the API. I still think that the tareget virt-handler know very well that the VMI it just received is from a migration (we have the post migration handling already). Can't you just pass there something to virt-launcher to sync? It really does not look like something to me which requires something stored in the status.

Reply:
@rmohr can you please point me to the post-migration handling code? 
I know we have a post-migration source cleanup, but I'm not sure how does the destination virt-handler know that migration has just completed...

Reply:
We don't directly have it in the code where we manage the domain, but we have an example in the status calculation. This snippet:

```
		if domainExists && vmi.Status.MigrationState != nil {
			// record that we've see the domain populated on the target's node
			log.Log.Object(vmi).Info("The target node received the migrated domain")
			vmiCopy.Status.MigrationState.TargetNodeDomainDetected = true
		}
```
is executed by the receiving virt-handler when it sees the domain running on its side.

I think that this information is enough to caluclate on the corresponding other place if a clock sync needs to be performed.

Reply:
Yes, I see what you mean. The trouble with this is that we want to call `setTime` only once after the migration is completed.
`if domainExists && vmi.Status.MigrationState != nil && vmi.Status.MigrationState.Completed` also isn't enough to guarantee that and we need another indication somewhere...

Reply:
I don't think that the status can give you that guarantee. If the connection is broken at the wrong moment it may try very often to set the time again. If that is what we need, I would suggest to record it in virt-launcher or a temporary file somehow.